lib/SimpleHttpClient/Communicator.cpp

```diff
@@ -445,12 +445,10 @@ void Communicator::handleResult(CURL* handle, CURLcode rc) {
     (*rip->_options._curlRcFn)(rc);
   }
 
+  MUTEX_LOCKER(guard, _handlesLock);
```

Comment:
I think this is now a bit of an over-pessimization, as the mutex will be held while `_scheduleMe` is called and the variables for the closure will be copied into the it. Anyway, it looks safe, and I don't think it will be a huge bottleneck. If it turns out to be one, we can make the mutex just cover the find and erase operations, and the release it for calling `_scheduleMe`.